src/tensors/levels/denselevels.jl

```diff
@@ -34,8 +34,10 @@ DenseLevel{Ti}(lvl::Lvl, shape) where {Ti, Lvl} = DenseLevel{Ti, Lvl}(lvl, shape
 const Dense = DenseLevel
 
 Base.summary(lvl::Dense) = "Dense($(summary(lvl.lvl)))"
-similar_level(lvl::DenseLevel) = Dense(similar_level(lvl.lvl))
-similar_level(lvl::DenseLevel, dims...) = Dense(similar_level(lvl.lvl, dims[1:end-1]...), dims[end])
+
+similar_level(lvl::DenseLevel, fill_value, eltype::Type) = Dense(similar_level(lvl.lvl, fill_value, eltype))
```

Comment:
This one should use the pre-existing dimension in lvl.

Reply:
alternately, you could delete these versions of similar_level and only use the one that takes dims

Reply:
Ok, I tried to remove `similar_level` functions without `dims` argument, but I'm not sure if we can do this. There's one place here:
https://github.com/willow-ahrens/Finch.jl/blob/11cecd0641d2d408463be3ec3617aa232732c8a3/src/tensors/levels/separatelevels.jl#L131
where I don't know what `dims` I should pass. How can I know the dimensionality of `lvl`?
I tried:
```julia
$sym = similar_level($(lvl.ex).lvl, $(virtual_level_default(lvl)), $(virtual_level_eltype(lvl)), $(virtual_level_size(lvl, ctx)))

# and different variations of the last argument:
$(virtual_level_size(lvl, ctx)...)
$(virtual_level_size(lvl, ctx))...
```
But none of them compile/expand macro without an error. Maybe I could treat the remaining levels as zero size with `[0]` repeated `level_ndims(lvl)` times?

Reply:
Oh no! Yes, okay, in that situation you could pass in `level_size`

Reply:
actually, the appropriate thing to do would be (I know this is arcane)

```
$sym = similar_level(
    $(lvl.ex).lvl,
    $(ctx(virtual_level_default(lvl))),
    $(virtual_level_eltype(lvl)),
    $(map(ctx, map(getstop, virtual_level_size(lvl, ctx)))...)
)
```

Reply:
Since you have the level already constructed you could also do:
```
$sym = similar_level(
    $(lvl.ex).lvl,
    level_default($(lvl.ex).lvl),
    level_eltype($(lvl.ex).lvl),
    level_size($(lvl.ex).lvl)
)
```

Reply:
Thank you!
The final version that worked is:
```
$sym = similar_level(
    $(lvl.ex).lvl,
    level_default(typeof($(lvl.ex).lvl)),
    level_eltype(typeof($(lvl.ex).lvl)),
    level_size($(lvl.ex).lvl)...
)
```

Reply:
I think thats the cleanest in that situation